src/program/frame.rs

```diff
@@ -87,3 +126,27 @@ impl FrameSet {
             .collect()
     }
 }
+
+pub(crate) enum FrameMatchCondition<'a> {
+    /// Match all frames in the set
+    All,
+
+    /// Match all frames which shares any one of these names
+    AnyOfNames(&'a [String]),
+
+    /// Match all frames which contain any of these qubits
+    AnyOfQubits(&'a [Qubit]),
+
+    /// Match all frames which contain exactly these qubits
+    ExactQubits(&'a [Qubit]),
+
+    /// Return these specific frames, if present in the set
+    Specific(&'a FrameIdentifier),
+
+    /// Return all frames which match all of these conditions
+    And(Vec<FrameMatchCondition<'a>>),
+
+    /// Return all frames which match any of these conditions
+    #[allow(dead_code)]
```

Comment:
Why does dead code need to be allowed here? I see `FrameMatchCondition::Or` above on line 78.

Reply:
But it's never constructed anywhere I think is the point. Similar to how if you have `f(g(x))` and `f()` is never called, `g()` is considered dead.

Reply:
Can we remove it then? Or is it here for future-proofing?

Reply:
Removed